src/utils/fetch.ts

```diff
@@ -1,23 +1,31 @@
-import Promise from 'promise-polyfill';
-import { hasWin } from './mixins';
+import Promise from "promise-polyfill";
+import { hasWin } from "./mixins";
 
 if (hasWin()) {
   window.Promise = window.Promise || Promise;
 }
 
-export default typeof fetch == 'function'
-  ? fetch.bind()
-  : (url, options) => {
-      return new Promise((res, rej) => {
+interface FetchOptions {
+  method: "POST" | "GET";
+  credentials: string;
+  headers?: { [key: string]: string };
+  onProgress: ((this: XMLHttpRequest, ev: ProgressEvent) => any) | null;
+  body?: Document | BodyInit | null;
```

Comment:
I believe this should be changed to `Document | XMLHttpRequestBodyInit | null`. 

This option is being passed to `XMLHttpRequest.send(body?: Document | XMLHttpRequestBodyInit | null)` on line 42. `BodyInit` is a union type of `ReadableStream` and `XMLHttpRequestBodyInit`, so `BodyInit` is not assignable to `XMLHttpRequestBodyInit`.